src/ui/mainwindow.cpp

```diff
@@ -85,6 +85,40 @@ struct TabState
 #endif
     }
 
+    TabState(const TabState &rhs)
+        : searchQuery(rhs.searchQuery)
+        , selections(rhs.selections)
+        , expansions(rhs.expansions)
+        , searchScrollPosition(rhs.searchScrollPosition)
+        , tocScrollPosition(rhs.tocScrollPosition)
+        , webViewZoomFactor(rhs.webViewZoomFactor)
+    {
+        searchModel = new Zeal::SearchModel();
+        tocModel = new Zeal::SearchModel();
+
+        webPage = new QWebPage();
+#ifndef USE_WEBENGINE
+        webPage->setLinkDelegationPolicy(QWebPage::DelegateExternalLinks);
+        webPage->setNetworkAccessManager(Core::Application::instance()->networkManager());
+#endif
+
+        QByteArray historyArray;
+        rhs.saveHistory(historyArray);
+        restoreHistory(historyArray);
+    }
+
+    void restoreHistory(const QByteArray &data) const
+    {
+        QDataStream stream(data);
+        stream >> *webPage->history();
+    }
+
+    void saveHistory(QByteArray &data) const
```

Comment:
Just for the sake of consistency, let's follow Qt's pattern (see `QSplitter::saveState()`, `QByteArray QWidget::saveGeometry()`) here:

```cpp
    QByteArray saveHistory() const
    {
        QByteArray history;
        QDataStream stream(&history, QIODevice::WriteOnly);
        stream << *webPage->history();
        return history;
    }
```